### PR TITLE
fix: [IA-90] The traits of some links are not read correctly

### PR DESCRIPTION
**Pull request by ncannata-dev**
## Short description
With the merge of this [PR](https://github.com/pagopa/io-app/commit/0c4dbe7a78982535bc718f88080def01a2279193), the trait of all link components is read correctly, but there are still many links in the application that are actually managed in the code as native-base legacy components

## List of changes proposed in this pull request
As a solution for this story, I converted all those links that were basically managed at the end as a <Text link = {true} /> into real <Link /> so as to have more uniformity and cleanliness at the code level

### CODE REVIEW

File: ts/screens/wallet/payment/ConfirmPaymentMethodScreen.tsx [hunk 1]
```diff
@@ -251,14 +252,16 @@ const ConfirmPaymentMethodScreen: React.FC<Props> = (props: Props) => {
               <H4>{` ${maybePsp.value.businessName}`}</H4>
             </H4>
           )}
-          <TouchableDefaultOpacity onPress={props.pickPsp}>
-            <Text link={true} bold={true}>
+          <TouchableDefaultOpacity accessible>
```
Comment (pietro909) [inline ts/screens/wallet/payment/ConfirmPaymentMethodScreen.tsx:255]: Do you still need the `TouchableDefaultOpacity` now that the press event is handled by the inner `Link`?

Comment (ncannata-dev) [reply ts/screens/wallet/payment/ConfirmPaymentMethodScreen.tsx]: Fixer PR has suggested ✅

File: ts/screens/wallet/TransactionDetailsScreen.tsx [hunk 2]
```diff
@@ -221,8 +222,10 @@ class TransactionDetailsScreen extends React.Component<Props, State> {
             recipient={transaction.merchant}
             description={cleanTransactionDescription(transaction.description)}
           />
-          <TouchableWithoutFeedback onPress={this.handleOnFullReasonPress}>
-            <Text link={true}>{I18n.t("wallet.transactionFullReason")}</Text>
+          <TouchableWithoutFeedback>
```
Comment (pietro909) [inline ts/screens/wallet/TransactionDetailsScreen.tsx:225]: This wrapper is redundant as well.

File: ts/screens/authentication/LandingScreen.tsx [hunk 3]
```diff
@@ -293,17 +294,15 @@ class LandingScreen extends React.PureComponent<Props, State> {
...
           <View spacer={true} />
 
-          <ButtonDefaultOpacity
-            block={true}
-            small={true}
-            transparent={true}
-            onPress={this.navigateToSpidCieInformationRequest}
-          >
-            <Text style={styles.noPadded} link={true}>
+          <ButtonDefaultOpacity block={true} small={true} transparent={true}>
```
Comment (pietro909) [inline ts/screens/authentication/LandingScreen.tsx:297]: This is redundant as well, but mind the style! Once you remove it, the link will lose some formatting.
Please check it.

with:
Screenshot 2021-07-22 at 17 03 09

without:
Screenshot 2021-07-22 at 17 03 01

**Review by pietro909 — CHANGES_REQUESTED** [review]
Please @ncannata-dev address the comments. They are all a special case of my initial comment. Now I pointed them out one by one and even tested locally to narrow down possible issues.

File: ts/screens/wallet/TransactionDetailsScreen.tsx (hunk 2 above)
Comment (ncannata-dev) [reply ts/screens/wallet/TransactionDetailsScreen.tsx]: Fixed PR as suggested ✅

File: ts/screens/wallet/payment/ManualDataInsertionScreen.tsx [hunk 4]
```diff
@@ -144,8 +145,10 @@ class ManualDataInsertionScreen extends React.Component<Props, State> {
           <Content scrollEnabled={false}>
             <H1>{I18n.t("wallet.insertManually.title")}</H1>
             <Text>{I18n.t("wallet.insertManually.info")}</Text>
-            <TouchableDefaultOpacity onPress={this.showModal}>
-              <Text link={true}>{I18n.t("wallet.insertManually.link")}</Text>
+            <TouchableDefaultOpacity>
```
[the remark this answers is not in the document]
Comment (ncannata-dev) [reply ts/screens/wallet/payment/ManualDataInsertionScreen.tsx]: Fixed PR as suggested ✅

File: ts/screens/authentication/LandingScreen.tsx (hunk 3 above)
Comment (ncannata-dev) [reply ts/screens/authentication/LandingScreen.tsx]: Fixed PR as suggested ✅
I removed the unused portion of code and fixed the style of the link component, now it is identical to how it was before the change

**Review by pietro909 — APPROVED** [review]

**Pull request merged**